### PR TITLE
feat: add completion

### PR DESCRIPTION
**Pull request by henry-hsieh**
This PR uses clap_complete to generate the completion script. To generate the completion script, use `tinty --generate SHELL > script_path`.

The reason why I added generated complete scripts to the Git index is that I want the schemes can be selected in completion directly. Currently, clap_complete cannot generate the complete results dynamically based on the outcome of another command, just like `tinty apply $(tinty list | fzf)` trick. Therefore, I have to manually change some places of the generated scripts.

However, I only make changes on bash only. The zsh, fish, and ps1 are untouched in this commit.

Here is the sample after `source tinty.bash` and hit Tab for completion:
圖片
Using [FZF Tab Completion](https://github.com/lincheney/fzf-tab-completion)
圖片

### CODE REVIEW

**Review by JamyGolden — CHANGES_REQUESTED** [review]
Thanks for the PR, I was planning on adding completions so this is very helpful.

- Since you've done some changes to the `.bash` completion file, I'd like some changes to the PR to make sure it's easily visible in git (and I'll do a rebase to include both commits into main). Could you squash all of your changes into 1 commit with no changes to the `contrib/completions/tinty.bash` file and then add another commit which adds the changes to the bash file? These files will need to be updated in future when new commands are added and it would help a lot to be able to see the diff which includes your completion additions so we can make sure to include them.
- This PR is missing docs and tests. It's ok though I can always add it after merge but before the next release so everything is included.
- I've added a comment on the line itself, but I suggest changing the command `generate` to something else.

File: src/cli.rs [hunk 1]
```diff
@@ -26,6 +27,13 @@ fn build_cli() -> Command {
                 .global(true)
                 .action(ArgAction::Set)
         )
+        .arg(
+            Arg::new("generate")
```
[the remark this answers is not in the document]
Comment (henry-hsieh) [reply src/cli.rs:31]: That's good. I'll change it to `generate-completion`.

**Review by JamyGolden — APPROVED** [review]
Looks great, let me know if I can rebase - I don't want to do that if you're planning on adding a last minute change :sweat_smile:

**Comment by henry-hsieh** [comment]
I'm done with changes. Please go ahead.

**Pull request merged**